public/app/plugins/datasource/loki/querybuilder/components/QueryPatternsModal.tsx

```diff
@@ -0,0 +1,176 @@
+import { css } from '@emotion/css';
+import React, { useRef, useState } from 'react';
+
+import { DataQuery, GrafanaTheme2 } from '@grafana/data';
+import { Button, Card, Collapse, Modal, useStyles2 } from '@grafana/ui';
+import { getNextRefIdChar } from 'app/core/utils/query';
+import { RawQuery } from 'app/plugins/datasource/prometheus/querybuilder/shared/RawQuery';
+
+import logqlGrammar from '../../syntax';
+import { LokiQuery } from '../../types';
+import { lokiQueryModeller } from '../LokiQueryModeller';
+import { buildVisualQueryFromString } from '../parsing';
+import { LokiQueryPattern, LokiQueryPatternType } from '../types';
+
+type Props = {
+  isOpen: boolean;
+  query: LokiQuery;
+  queries: DataQuery[] | undefined;
+  onClose: () => void;
+  onChange: (query: LokiQuery) => void;
+  onAddQuery?: (query: LokiQuery) => void;
+};
+
+export const QueryPatternsModal = (props: Props) => {
+  const { isOpen, onClose, onChange, onAddQuery, query, queries } = props;
+  const [openTabs, setOpenTabs] = useState<string[]>([]);
+  const [selectedPatternConfirmationIndex, setSelectedPatternConfirmationIndex] = useState<number | null>(null);
+  const indexOfSelectedCard = useRef<number | null>(null);
+
+  const styles = useStyles2(getStyles);
+  const lang = { grammar: logqlGrammar, name: 'logql' };
+  const hasPreviousQuery = buildVisualQueryFromString(query.expr).query.operations.length > 0;
+  const hasNewQueryOption = !!onAddQuery;
+
+  const onPatternSelect = (pattern: LokiQueryPattern, selectAsNewQuery = false) => {
+    const queryString = selectAsNewQuery ? '' : query.expr;
+    const visualQuery = buildVisualQueryFromString(queryString);
+    visualQuery.query.operations = pattern.operations;
+
+    if (hasNewQueryOption && selectAsNewQuery) {
+      onAddQuery({
+        ...query,
+        refId: getNextRefIdChar(queries ?? [query]),
+        expr: lokiQueryModeller.renderQuery(visualQuery.query),
+      });
+    } else {
+      onChange({
+        ...query,
+        expr: lokiQueryModeller.renderQuery(visualQuery.query),
+      });
+    }
+    setSelectedPatternConfirmationIndex(null);
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} title="Kick start your query" onDismiss={onClose}>
+      <div className={styles.spacing}>
+        Kick start your query by selecting one of these queries. You can then continue to complete your query.
+      </div>
+      {Object.values(LokiQueryPatternType).map((patternType) => {
+        return (
+          <Collapse
+            key={patternType}
+            label={`${patternType[0].toUpperCase() + patternType.slice(1)} query starters`}
```

Comment:
Would it look too bad if we replace this with CSS `text-transform: capitalize`?

Reply:
Yeah, I tried that, but we specify the text as `Collapse` `label`. And if we add styling to `Collapse` it affects everything in it - cards, code,... . But I got other idea - to just use `lodash` `capitalize` to make this cleaner and readable.

Reply:
updated